documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,645 @@
+
+Upon the arrival of new flows, whether individually or in batches, corresponding flow entries are added to the table. These entries may represent either bidirectional or unidirectional flows. For bidirectional flows, the implementation adds entries for both the original flow and its reverse, linking their reverse flow keys to each other. For unidirectional flows, the current direction is specified. If a reverse flow for a unidirectional flow is created later, the user can add reverse keys for both and link them accordingly.
+
+Flows can be modified and removed through the DASH flow SAI API and can also be aged by the hardware.
+
+For more use cases, please refer to [Smart Switch HA HLD](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/smart-switch-ha-hld.md).
+
+## Flow Table APIs
+
+The flow_table APIs are define as follows:
+
+| API                      | Description                           |
+| ------------------------ | :------------------------------------ |
+| create_flow_table        | Create a new flow table               |
+| remove_flow_table        | Remove a flow table                   |
+| set_flow_table_attribute | Set the attributes of a flow table    |
+| get_flow_table_attribute | Obtain the attributes of a flow table |
+| create_flow_tables       | Create multiple flow tables in bulk   |
+| remove_flow_tables       | Remove multiple flow tables in bulk   |
+
+The attributes of the flow_table are defined as follows:
+
+| Attribute name                         | Type                       | Description                                     |
+|----------------------------------------|----------------------------|-------------------------------------------------|
+| SAI_FLOW_TABLE_ATTR_MAX_FLOW_COUNT     | `sai_uint32_t`             | Maximum number of flows allowed in the table.   |
+| SAI_FLOW_TABLE_ATTR_DASH_FLOW_ENABLED_KEY | `sai_dash_flow_enabled_key_t` | Key enable mask |
+| SAI_FLOW_TABLE_ATTR_FLOW_TTL_IN_MILLISECONDS | `sai_uint32_t`     | Time-to-live (TTL) for flows in milliseconds.  |
+
+The sai_dash_flow_enabled_key_t is defined as follows:
+
+```c
+typedef enum _sai_dash_flow_enabled_key_t
+{
+    SAI_DASH_FLOW_ENABLED_KEY_NONE = 0,
+
+    SAI_DASH_FLOW_ENABLED_KEY_ENI_ADDR = 1 << 1,
+
+    SAI_DASH_FLOW_ENABLED_KEY_PROTOCOL = 1 << 2,
+
+    SAI_DASH_FLOW_ENABLED_KEY_SRC_IP = 1 << 3,
+
+    SAI_DASH_FLOW_ENABLED_KEY_DST_IP = 1 << 4,
+
+    SAI_DASH_FLOW_ENABLED_KEY_SRC_PORT = 1 << 5,
+
+    SAI_DASH_FLOW_ENABLED_KEY_DST_PORT = 1 << 6,
+
```

Comment:
In NAT scenario, SAI_DASH_FLOW_ENABLED_KEY_VNI is needed, would you please help add this field support? Thanks.

Reply:
Thanks, added.